sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_call_automation_client.py

```diff
@@ -426,31 +498,34 @@ def start_recording(
         :rtype: ~azure.communication.callautomation.RecordingProperties
         :raises ~azure.core.exceptions.HttpResponseError:
         """
-        channel_affinity_internal = []
 
-        if channel_affinity:
-            for channel in channel_affinity:
-                channel_affinity_internal.append(channel._to_generated(# pylint:disable=protected-access
-                    ))
+    @distributed_trace
+    def start_recording(
+        self,
+        *args: Union['ServerCallLocator', 'GroupCallLocator'],
+        **kwargs
+    ) -> RecordingProperties:
+        # pylint:disable=protected-access
+        channel_affinity: List[ChannelAffinity] = kwargs.pop("channel_affinity", None) or []
+        channel_affinity_internal = [c._to_generated() for c in channel_affinity]
+        call_locator = build_call_locator(args, kwargs)
 
         start_recording_request = StartCallRecordingRequest(
-            call_locator=call_locator._to_generated(# pylint:disable=protected-access
-            ),
-            recording_state_callback_uri = recording_state_callback_url,
-            recording_content_type = recording_content_type,
-            recording_channel_type = recording_channel_type,
-            recording_format_type = recording_format_type,
-            audio_channel_participant_ordering = audio_channel_participant_ordering,
-            recording_storage_type = recording_storage_type,
-            external_storage_location = external_storage_location,
+            call_locator=call_locator,
+            recording_state_callback_uri = kwargs.pop("recording_state_callback_url", None),
+            recording_content_type = kwargs.pop("recording_content_type", None),
+            recording_channel_type = kwargs.pop("recording_channel_type", None),
+            recording_format_type = kwargs.pop("recording_format_type", None),
+            audio_channel_participant_ordering = kwargs.pop("audio_channel_participant_ordering", None),
+            recording_storage_type = kwargs.pop("recording_storage_type", None),
+            external_storage_location = kwargs.pop("external_storage_location", None),
```

Comment:
nit spacing

```suggestion
            recording_state_callback_uri=kwargs.pop("recording_state_callback_url", None),
            recording_content_type=kwargs.pop("recording_content_type", None),
            recording_channel_type=kwargs.pop("recording_channel_type", None),
            recording_format_type=kwargs.pop("recording_format_type", None),
            audio_channel_participant_ordering=kwargs.pop("audio_channel_participant_ordering", None),
            recording_storage_type=kwargs.pop("recording_storage_type", None),
            external_storage_location=kwargs.pop("external_storage_location", None),
```

Reply:
Shouldn't them be caught by black?

Reply:
Do you mean in CI? I think that check is opt-in.

Reply:
Oh, I thought it was on by default.